modules/core/src/main/java/org/apache/ignite/internal/IgniteInternalFuture.java

```diff
@@ -132,6 +132,18 @@
     @Async.Schedule
     public <T> IgniteInternalFuture<T> chain(IgniteClosure<? super IgniteInternalFuture<R>, T> doneCb, Executor exec);
 
+    /**
+     * Make a chained future that is completed when {@code doneCb} is executed. Callback is called with this future
+     * as the argument, when this future completes. It is guaranteed that done callback will be called only ONCE.
+     *
+     * @param doneCb Done callback.
+     * @param <T> Type parameter.
+     * @return Chained future.
+     */
+    public <T> IgniteInternalFuture<T> chainCompose(
```

Comment:
I think it would be nice to do the following things:
 - annotate this method with the `@Async.Schedule` annotation 
 - add an additional version of this method with a custom executor